### PR TITLE
PP-10713 implement idempotency key header violation

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- implement checking the length of the Idempotency-Key header if present.
- refactor RequestError to include a `header` field
- refactor ViolationExceptionMapper to handle this violation
- add header param to PaymentsResource create payment
- add/update relevant tests

with @alexbishop1

### CODE REVIEW

File: openapi/publicapi_spec.json [hunk 1]
```diff
@@ -2660,6 +2667,11 @@
             "type" : "string",
             "description" : "The parameter in your request that's causing the error.",
             "example" : "amount"
+          },
+          "header" : {
+            "type" : "string",
+            "description" : "The header in your request that's causing the error.",
```
Comment (stephencdaly) [inline openapi/publicapi_spec.json:2673]: Worth double checking this wording with Nathan if you haven't already

File: src/main/java/uk/gov/pay/api/model/RequestError.java [hunk 2]
```diff
@@ -127,26 +129,36 @@ public static RequestError aRequestError(Code code, Object... parameters) {
     }
 
     public static RequestError aRequestError(String fieldName, Code code, Object... parameters) {
-        return new RequestError(fieldName, code, parameters);
+        return new RequestError(null, fieldName, code, parameters);
+    }
+
+    public static RequestError aRequestErrorHeader(String header, Code code, Object... parameters) {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/model/RequestError.java:135]: would `aHeaderRequestError` be a better name - as it's not a header that's being returned

**Review by stephencdaly — APPROVED** [review]
Looks good, very minor comment about a method name

File: openapi/publicapi_spec.json (hunk 1 above)
Comment (SandorArpa) [reply openapi/publicapi_spec.json:2673]: 👍

File: src/main/java/uk/gov/pay/api/model/RequestError.java (hunk 2 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/api/model/RequestError.java:135]: 👍

**Review by stephencdaly — APPROVED** [review]

**Pull request merged**